app/routes.py

```diff
@@ -0,0 +1,444 @@
+    return make_response(new_rental.to_dict(checked_out=video.inventory_checked_out, available_inventory=inventory_available), 200)
+
+def process_checkin(request_body):
+    video_id = request_body["video_id"]
+    customer_id = request_body["customer_id"]
+    video = Video.query.get(video_id)
+    video.inventory_checked_out -= 1
+    inventory_available = calculate_inventory_available(video)
+    rentals = Rental.query.filter(Rental.customer_id==customer_id).all() #
+    result = make_response({'message': f"No outstanding rentals for customer {customer_id} and video {video_id}"}, 400)
+    if not rentals:
+        return result  
+    for rental in range(len(rentals)):
+        if rentals[rental].video_id == video_id:
+            result = make_response(rentals[rental].to_dict(checked_out=video.inventory_checked_out, available_inventory=inventory_available), 200)
+            db.session.delete(rentals[rental])
+            db.session.commit()
+    return result    
+
+#####---------------------------    Sorting   -------------------------------#####
+# Customers can be sorted by name, registered_at and postal_code
+# Videos can be sorted by title and release_date
+# Overdue rentals can be sorted by title, name, checkout_date and due_date
+
+
+def sort_titles(sort_by, entity):
+    #Thinking about making this a very generic function to sort anything with a simple order_by 
+    if sort_by == "asc": 
+        sorted = entity.query.order_by(entity.title.asc())
+    elif sort_by == "desc":
+        sorted = entity.query.order_by(entity.title.desc())
+    else:
+        sorted = entity.query.all()
+    return sorted
+
+def sort_alpha(sort_by, entity, category): #Rename to: sort_alpha?
+    #Thinking about making this a very generic function to sort anything with a simple order_by 
+    #.limit()
+
+    if sort_by == "asc": 
+        sorted = entity.query.order_by(entity.category.asc())
+    elif sort_by == "desc":
+        sorted = entity.query.order_by(entity.category.desc())
+    else:
+        sorted = entity.query.all()
+    return sorted
+
+def sort_dates(sort_by, entity, category):
+    #May want to use this for release_dates if sort_titles can't be made generic?
+    if sort_by == "asc":
+        sorted = entity.query.order_by(entity.category.asc())
+    elif sort_by == "desc":
+        sorted = entity.query.order_by(entity.category.desc())
+    else:
+        sorted = entity.query.all()
+    return sorted
+
+    # ex: entities = MyEntity.query.order_by(desc(MyEntity.time)).limit(3).all()
+
+#####------------------------    Data Checking   ----------------------------#####
+def check_customer_video_data(method=None, request_body=None, keys=None, id=None, model=None, entity=None):
+    if method == "POST" or method == "PUT":
+        for key in keys:
+            if key not in request_body.keys():
+                return make_response({"details": f"Request body must include {key}."}, 400)
```

Comment:
Using abort here would could simplify the code around this function:
```suggestion
                abort(make_response({"details": f"Request body must include {key}."}, 400))
```
The calling function could then just call check_customer_video_data(...) rather than capturing and testing the return value from the function.